polaris-react/src/components/Text/Text.scss

```diff
@@ -57,67 +57,67 @@
 }
 
 @mixin _headingSm {
-  font-size: var(--p-font-size-80-experimental);
-  line-height: var(--p-font-line-height-2);
+  font-size: var(--p-text-heading-sm-font-size);
+  line-height: var(--p-text-heading-sm-font-line-height);
 }
 
 .headingXs {
-  font-weight: var(--p-font-weight-semibold);
+  font-weight: var(--p-text-heading-xs-font-weight);
   @include _headingSm;
 }
 
 .headingSm {
-  font-weight: var(--p-font-weight-semibold);
+  font-weight: var(--p-text-heading-sm-font-weight);
   @include _headingSm;
 }
 
 .headingMd {
-  font-size: var(--p-font-size-100);
-  line-height: var(--p-font-line-height-2);
-  font-weight: var(--p-font-weight-semibold);
+  font-size: var(--p-text-heading-md-font-size);
+  line-height: var(--p-text-heading-md-font-line-height);
+  font-weight: var(--p-text-heading-md-font-weight);
 }
 
 .headingLg {
-  font-size: var(--p-font-size-300);
-  line-height: var(--p-font-line-height-3);
+  font-size: var(--p-font-size-500);
+  line-height: var(--p-font-line-height-600);
   font-weight: var(--p-font-weight-bold);
 
   @media #{$p-breakpoints-md-up} {
-    font-size: var(--p-font-size-300);
-    line-height: var(--p-font-line-height-600);
+    font-size: var(--p-text-heading-lg-font-size);
+    line-height: var(--p-text-heading-lg-font-line-height);
   }
 }
 
 .headingXl {
-  font-size: var(--p-font-size-300);
-  line-height: var(--p-font-line-height-3);
+  font-size: var(--p-font-size-500);
+  line-height: var(--p-font-line-height-600);
   font-weight: var(--p-font-weight-bold);
 
   @media #{$p-breakpoints-md-up} {
-    font-size: var(--p-font-size-400);
-    line-height: var(--p-font-line-height-5);
+    font-size: var(--p-text-heading-xl-font-size);
+    line-height: var(--p-text-heading-xl-font-line-height);
   }
 }
 
 .heading2xl {
-  font-size: var(--p-font-size-400);
-  line-height: var(--p-font-line-height-5);
+  font-size: var(--p-font-size-600);
+  line-height: var(--p-font-line-height-800);
   font-weight: var(--p-font-weight-bold);
 
   @media #{$p-breakpoints-md-up} {
-    font-size: var(--p-font-size-500);
-    line-height: var(--p-font-line-height-6);
+    font-size: var(--p-text-heading-2xl-font-size);
+    line-height: var(--p-text-heading-2xl-font-line-height);
```

Comment:
I only added the heading2xl composite tokens for `breakpoints-md-up` since responsiveness was out of scope when these tokens were created.